widgets/common/src/filters/FiltersDefinitionForm.tsx

```diff
@@ -0,0 +1,34 @@
+import { useState } from 'react';
+import type { FunctionComponent } from 'react';
+
+import Form from './FormWrapper';
+import RuleRow from './RuleRow';
+import AddRuleButton from './AddRuleButton';
+
+const defaultRow = { attribute: '', operator: '', input: '' };
+
+const FiltersDefinitionForm: FunctionComponent = () => {
+    const [rows, setRows] = useState([defaultRow]);
+
+    function addRule() {
+        const newRows = [...rows, defaultRow];
+        setRows(newRows);
```

Comment:
We could use the function form of the updater function, just in case :slightly_smiling_face: 

```suggestion
        setRows((rows) => ([...rows, defaultRow]));
```

Reply:
Oh, I wasn't aware of that form, updated 👍🏻